### PR TITLE
Django webserver and basic webapp

### PR DESCRIPTION
**Pull request by avdonovan**
**1. Brief Summary of what this PR accomplishes (140 characters or less. If you find trouble describing what you are doing in this length, consider breaking the PR into multiple ones.)**

Adds a Django project with basic API endpoint and basic webapp.

**2. Link to Trello Ticket**

https://trello.com/c/IaWsXnHr/21-web-server

**3. More detailed description and other questions to address in code review**

**4. Remember to tag reviewers!**

### CODE REVIEW

**Comment by avdonovan** [comment]
I also added the conda-forge channel to the anaconda env to get the latest version of Django... not sure if this breaks anything on the datascience side so someone should try updating their env on this branch to confirm everything's working before merging in.

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 *.pyc
 Campaign_Finance_-_FPPC_Form_460_-_Schedule_A_-_Monetary_Contributions.csv
 notebooks/.ipynb_checkpoints/*
+.idea
```
Comment (erikeldridge) [inline .gitignore]: minor: seems like end-of-file newlines are preferred

File: django/congressionaldata/api/views.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (erikeldridge) [inline django/congressionaldata/api/views.py]: minor: remove boilerplate comment

File: django/congressionaldata/api/views.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+class CandidateContributionsViewSet(viewsets.ReadOnlyModelViewSet):
+
+    serializer_class = CandidateContributionSerializer
+
+    # override the default query set to limit to 100 objects
+    queryset = CandidateContributions.objects.all()[0:100]
+
+
+# TODO: viewsets for other models, and probably more endpoints
```
Comment (erikeldridge) [inline django/congressionaldata/api/views.py]: minor: this todo seems vague. Is there a known action to take? Also, associating todos w a ticket/person can help ensure follow-up, eg "TODO add view sets for all models: https://trello.com/c/IaWsXnHr/21-view-sets-for-all-models"

File: django/congressionaldata/congressionaldata/settings.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/2.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
```
Comment (erikeldridge) [inline django/congressionaldata/congressionaldata/settings.py]: Seems we should be pulling this from an env var and not checking it in.

File: django/congressionaldata/congressionaldata/settings.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+]
+
+WSGI_APPLICATION = 'congressionaldata.wsgi.application'
+
+
+# Database
+# https://docs.djangoproject.com/en/2.0/ref/settings/#databases
+
+DATABASES = {
+    # Ask on slack for credentials
```
Comment (erikeldridge) [inline django/congressionaldata/congressionaldata/settings.py]: Thinking out loud: this instruction is consistent w our onboarding (https://github.com/sfbrigade/datasci-congressional-data/blob/master/onboarding/02_development_environment.md#connecting-to-our-database).

File: django/congressionaldata/webapp/templates/webapp/index.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        <ul></ul>
+
+        <script
+          src="http://code.jquery.com/jquery-3.3.1.min.js"
+          integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8="
+          crossorigin="anonymous"></script>
+
+        <script type="text/javascript">
+            // TODO set up separate JS files and a real build process
+            $(function() {
```
Comment (erikeldridge) [inline django/congressionaldata/webapp/templates/webapp/index.html]: Nice, simple sample 👍

File: onboarding/04_01_running_the_webapp.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Running The Webapp
+
+## Webapp structure
+
+The Django project resides in the [django/congressionaldata](../django/congressionaldata) folder of the repository.  It contains two Django apps:
+
+* [api](../django/congressionaldata/api) is a JSON API server, which has mappings for the database tables in [models.py](../django/congressionaldata/api/models.py) and uses the Django Rest Framework to power the API (see [views.py](../django/congressionaldata/api/views.py) and [serializers.py](../django/congressionaldata/api/serializers.py)).
+* [webapp](../django/congressionaldata/webapp) is a static webapp that pulls data from the API server and renders a visualization.  It doesn't have any backend models, just a simple [view](../django/congressionaldata/webapp/views.py) and [HTML page](../django/congressionaldata/webapp/templates/webapp/index.html) and soon there'll be a lot af JavaScript.
```
Comment (erikeldridge) [inline onboarding/04_01_running_the_webapp.md]: minor: typo "af" -> "of"

File: .gitignore (hunk 1 above)
Comment (avdonovan) [reply .gitignore]: done

File: django/congressionaldata/api/views.py (hunk 2 above)
Comment (avdonovan) [reply django/congressionaldata/api/views.py]: done

File: django/congressionaldata/api/views.py (hunk 3 above)
Comment (avdonovan) [reply django/congressionaldata/api/views.py]: No specific action, just wanted to indicate that this is where we'd add a lot more stuff for the real app

File: django/congressionaldata/congressionaldata/settings.py (hunk 4 above)
Comment (avdonovan) [reply django/congressionaldata/congressionaldata/settings.py]: We can continue to use this for local development.  I added a check for an environment variable so we can override it on prod.

File: django/congressionaldata/congressionaldata/settings.py (hunk 5 above)
Comment (avdonovan) [reply django/congressionaldata/congressionaldata/settings.py]: updated

File: onboarding/04_01_running_the_webapp.md (hunk 7 above)
Comment (avdonovan) [reply onboarding/04_01_running_the_webapp.md]: fixed

File: django/congressionaldata/congressionaldata/settings.py (hunk 4 above)
Comment (erikeldridge) [reply django/congressionaldata/congressionaldata/settings.py]: 👍

Comment (erikeldridge) [reply django/congressionaldata/congressionaldata/settings.py]: sounds good. thx

**Comment by erikeldridge** [comment]
@avdonovan if I understand our process correctly, I think think you're free to merge this change now that it's been reviewed, ie as opposed to some folks submitting PRs and others reviewing and merging.

File: django/congressionaldata/api/models.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    donor_committee_name = models.TextField(blank=True, null=True)
+    donor_committee_type = models.TextField(blank=True, null=True)
+    donor_committee_party = models.TextField(blank=True, null=True)
+
+    class Meta:
+        managed = False
+        db_table = 'candidate_contributions'
+
+
+class StgCandidateContributions(models.Model):
```
Comment (VincentLa) [inline django/congressionaldata/api/models.py]: This is probably fine, but just as a note, I was thinking about the structure of the DB and I was planning that the way to structure this is that everything in `trg_analytics` is what gets surfaced outwards. Things in `stg_analytics` are purely meant for internal transformations.

We can also talk more about the structure of the DB

Comment (avdonovan) [reply django/congressionaldata/api/models.py]: Makes sense. Yeah I assume we're going to be running `inspectdb` again every time we change the DB schema and can choose which models to pull in.

**Pull request merged**